### ISSUE TITLE
Совместимость с Joomla 5

### ISSUE THREAD

**Issue by 1inkin** [PavelSyomin/proofreader#7]
Здравствуйте. После обновления сайта с Joomla 4.4.2 до Joomla 5 на страницах сайта отображается ошибка
`Event onContentAfterDisplay only accepts String results.`

**Comment by PavelSyomin** [comment]
Здравствуйте! Давненько сюда не залезал… Любопытно, что уведомления тоже почему-то не приходили. Постараюсь посмотреть этот и другие issues в ближайшие пару недель.

**Comment by GorillaYa** [comment]
> Здравствуйте. После обновления сайта с Joomla 4.4.2 до Joomla 5 на страницах сайта отображается ошибка `Event onContentAfterDisplay only accepts String results.`

Согласен, та же ошибка появилась после обновления до joomla 5.1, пришлось пока отключить компонент

**Comment by GorillaYa** [comment]
> Здравствуйте! Давненько сюда не залезал… Любопытно, что уведомления тоже почему-то не приходили. Постараюсь посмотреть этот и другие issues в ближайшие пару недель.

Удалось ли решить данную проблему?

**Comment by PavelSyomin** [comment]
К сожалению, пока нет: не нашлось времени. Скорее всего, задача не самая простая, так как исходный компонент сам по себе довольно старый (первоначальный автор делал его под J1.5), и в нём куча легаси-кода, который для нормальной совместимости с J5 придётся переписывать с нуля. 

У J5 есть [backward compatilibily plugin](https://manual.joomla.org/migrations/44-50/compat-plugin/), который, судя по описанию, возвращает какую-то часть совместимости с J4, а под J4 данная версия Proofreader работает. Можете, пожалуйста, попробовать установить его и проверить, работает ли Proofreader? Если да, то пока помечу его на сайте и здесь как «работающий на J5 с плагином совместимости». Если нет, то придётся, видимо, запланировать серьёзное переписывание кода.

**Comment by GorillaYa** [comment]
> К сожалению, пока нет: не нашлось времени. Скорее всего, задача не самая простая, так как исходный компонент сам по себе довольно старый (первоначальный автор делал его под J1.5), и в нём куча легаси-кода, который для нормальной совместимости с J5 придётся переписывать с нуля.
> 
> У J5 есть [backward compatilibily plugin](https://manual.joomla.org/migrations/44-50/compat-plugin/), который, судя по описанию, возвращает какую-то часть совместимости с J4, а под J4 данная версия Proofreader работает. Можете, пожалуйста, попробовать установить его и проверить, работает ли Proofreader? Если да, то пока помечу его на сайте и здесь как «работающий на J5 с плагином совместимости». Если нет, то придётся, видимо, запланировать серьёзное переписывание кода.

К сожалению у меня в режиме совместимости, также выдает вышеописанную ошибку (

**Comment by PavelSyomin** [comment]
Понял, спасибо за тестирование! Тогда, видимо, придётся всё же серьёзно покопаться в коде. По срокам, к сожалению, сориентировать не могу, так как сейчас у меня много других дел.

**Comment by 1inkin** [comment]
Как я понимаю, ошибка заключается в том, что от функции onContentAfterDisplay ожидается ответ строкой, но там при одном из условий стоит `return false`. Если в файле /plugins/content/proofreader/proofreader.php в строке 92 заменить `false` на `null`, то ошибка на страницах сайта пропадает. Но не до конца уверен насколько это корректно. Более знающие люди надеюсь проверят.

**Comment by PavelSyomin** [comment]
@1inkin Очень похоже на то, что вы правы. Попробуйте, пожалуйста, вот эту версию расширения: 
[proofreader_2_2.zip](https://github.com/PavelSyomin/proofreader/files/15305157/proofreader_2_2.zip) Предварительно включите Backwards Compatibility Plugin (Поведение - Обратная совместимость), так как без него точно не заработает. С ним же вроде как заработало (у меня). Если и у вас тоже заработает, то выпущу обновление: конечно, это будет не полноценная версия под J5, а просто быстрый фикс бага, но это лучше, чем ничего.

**Comment by 1inkin** [comment]
@PavelSyomin да, вроде бы этот вариант работает

**Comment by PavelSyomin** [comment]
Отлично! Выпустил обновление, обновил также данные о совместмости на JED.
